xarray/plot/utils.py

```diff
@@ -23,6 +24,21 @@ def _load_default_cmap(fname='default_colormap.csv'):
     return LinearSegmentedColormap.from_list('viridis', cm_data)
 
 
+def import_seaborn():
+    '''import seaborn and handle deprecation of apionly module'''
+    with warnings.catch_warnings(record=True) as w:
+        warnings.simplefilter("always")
+        try:
+            import seaborn.apionly as sns
+            if w and issubclass(w[-1].category, UserWarning):
+                raise ImportError
+        except ImportError:
+            import seaborn as sns
+        finally:
+            warnings.resetwarnings()
+    return sns
+
+
```

Comment:
@shoyer - it turns out the build failures we were seeing were because we were catching all the warnings/errors that scipy suppresses during its import. This solution seems to work but also relies on last (of many) warnings being the one from seaborn.

Reply:
You would think warnings would follow normal scoping rules like exceptions, but apparently not.